lib/stringify.js

```diff
@@ -148,6 +150,10 @@ var stringify = function stringify(
 
     var adjustedPrefix = commaRoundTrip && isArray(obj) && obj.length === 1 ? prefix + '[]' : prefix;
 
+    if (isArray(obj) && obj.length === 0 && allowEmptyArrays) {
```

Comment:
```suggestion
    if (allowEmptyArrays && isArray(obj) && obj.length === 0) {
```
so the cheaper check goes first